pkg/apis/provisioning/v1alpha5/labels.go

```diff
@@ -27,6 +27,9 @@ var (
 	ArchitectureArm64    = "arm64"
 	OperatingSystemLinux = "linux"
 
+	// ValidTopologyKeys are the topology keys that Karpenter allows for topology spread and pod affinity/anti-affinity
+	ValidTopologyKeys = stringsets.NewString(v1.LabelHostname, v1.LabelTopologyZone, LabelCapacityType)
```

Comment:
Is there a reason we have to be aware of this?

Reply:
We haven't tested with anything else, I think it's ok for this PR with the possibility to expand it in the future.  It's always been this limited for topology spread (capacity type is a recent addition).  Need to figure out what the use case is for other labels since the provider needs to put them on the pod, there probably is a use now considering the exists operator for labels.

Reply:
I can imagine some, but we can punt on it.